pkg/rate/api_limiter.go

```diff
@@ -0,0 +1,790 @@
+			return fmt.Errorf("unable to parse duration '%s': %w", value, err)
+		}
+		p.MinWaitDuration = minWaitDuration
+	case "max-wait-duration":
+		maxWaitDuration, err := time.ParseDuration(value)
+		if err != nil {
+			return fmt.Errorf("unable to parse duration '%s': %w", value, err)
+		}
+		p.MaxWaitDuration = maxWaitDuration
+	case "estimated-processing-duration":
+		estProcessingDuration, err := time.ParseDuration(value)
+		if err != nil {
+			return fmt.Errorf("unable to parse duration '%s': %w", value, err)
+		}
+		p.EstimatedProcessingDuration = estProcessingDuration
+	case "auto-adjust":
+		switch strings.ToLower(value) {
+		case "true":
+			p.AutoAdjust = true
+		case "false":
+			p.AutoAdjust = false
+		default:
+			return fmt.Errorf("unknown boolean value '%s'", value)
+		}
+	case "parallel-requests":
+		parallel, err := strconv.ParseUint(value, 0, 64)
+		if err != nil {
+			return fmt.Errorf("unable to parse integer '%s': %w", value, err)
+		}
+		p.ParallelRequests = int(parallel)
+	case "min-parallel-requests":
+		minParallel, err := strconv.ParseUint(value, 0, 64)
+		if err != nil {
+			return fmt.Errorf("unable to parse integer '%s': %w", value, err)
+		}
+		p.MinParallelRequests = int(minParallel)
+	case "max-parallel-requests":
+		maxParallel, err := strconv.ParseUint(value, 0, 64)
+		if err != nil {
+			return fmt.Errorf("unable to parse integer '%s': %w", value, err)
+		}
+		p.MaxParallelRequests = int(maxParallel)
+	case "mean-over":
+		meanOver, err := strconv.ParseUint(value, 0, 64)
+		if err != nil {
+			return fmt.Errorf("unable to parse integer '%s': %w", value, err)
+		}
+		p.MeanOver = int(meanOver)
+	case "log":
+		switch strings.ToLower(value) {
+		case "true":
+			p.Log = true
+		case "false":
+			p.Log = false
+		default:
+			return fmt.Errorf("unknown boolean value '%s'", value)
+		}
+	case "delayed-adjustment-factor":
+		delayedAdjustmentFactor, err := strconv.ParseFloat(value, 64)
+		if err != nil {
+			return fmt.Errorf("unable to parse float '%s': %w", value, err)
+		}
+		p.DelayedAdjustmentFactor = delayedAdjustmentFactor
+	case "max-adjustment-factor":
+		maxAdjustmentFactor, err := strconv.ParseFloat(value, 64)
+		if err != nil {
+			return fmt.Errorf("unable to parse float '%s': %w", value, err)
+		}
+		p.MaxAdjustmentFactor = maxAdjustmentFactor
+	case "skip-initial":
+		skipInitial, err := strconv.ParseUint(value, 0, 64)
+		if err != nil {
+			return fmt.Errorf("unable to parse integer '%s': %w", value, err)
+		}
+		p.SkipInitial = int(skipInitial)
+	default:
+		return fmt.Errorf("unknown rate limiting option '%s'", key)
+	}
+
+	return nil
+}
+
+func parseUserConfig(config string, p *APILimiterParameters) error {
+	tokens := strings.Split(config, ",")
+	for _, token := range tokens {
+		if token == "" {
+			continue
+		}
+
+		t := strings.SplitN(token, ":", 2)
+		if len(t) != 2 {
+			return fmt.Errorf("unable to parse rate limit option '%s', must in the form name=option:value[,option:value]", token)
```

Comment:
I think the function is doing the right thing for `name=option:value[,option:value]`, in that `parseUserConfig` would be passed the part after the `=` sign. However the tests don't seem to actually be testing the multi-value case.

Reply:
It was tested indirectly via `TestNewAPILimiterFromConfig` but I've added a direct test as well.